mwdb/web/src/types/types.ts

```diff
@@ -0,0 +1,211 @@
+    immutable: boolean;
+    name: string;
+    private: boolean;
+    users: string[];
+    workspace: boolean;
+};
+
+export type ObjectType = "file" | "blob" | "config";
+
+export type Attribute = {
+    key: string;
+    id: number;
+    value: unknown;
+};
+
+export type AttributeDefinition = {
+    description: string;
+    example_value: string;
+    hidden: boolean;
+    key: string;
+    label: string;
+    rich_template: string;
+    url_template: string;
+};
+
+export type Tag = {
+    tag: string;
+};
+
+export type RelatedObject = {
+    id: string;
+    tags: Tag[];
+    type: ObjectType;
+    upload_time: string | Date;
+};
+
+export type ConfigType = "static" | "dynamic";
+
+type ObjectCommonData = {
+    attributes: Attribute[];
+    children: RelatedObject[];
+    favorite: boolean;
+    id: string;
+    parents: RelatedObject[];
+    tags: Tag[];
+    upload_time: string | Date;
+};
+
+export type ObjectData = ObjectCommonData & {
+    alt_names: string[];
+    crc32: string;
+    file_name: string;
+    file_size: number;
+    file_type: string;
+    latest_config: null;
+    md5: string;
+    sha1: string;
+    sha256: string;
+    sha512: string;
+    ssdeep: string;
+    type: "file";
+};
+
+export type ConfigData = ObjectCommonData & {
+    cfg: object;
+    config_type: ConfigType;
+    family: string;
+    type: "static_config";
+};
+
+export type BlobData = ObjectCommonData & {
+    blob_name: string;
+    blob_size: number;
+    blob_type: string;
+    content: string;
+    last_seen: string | Date;
+    latest_config: unknown;
+    type: "text_blob";
+};
+
+export type ObjectListItem = {
+    file_name: string;
+    file_size: number;
+    file_type: string;
+    id: string;
+    md5: string;
+    sha1: string;
+    sha256: string;
+    tags: Tag[];
+    type: "file";
+    upload_time: string | Date;
+};
+
+export type ConfigListItem = {
+    config_type: ConfigType;
+    family: string;
+    id: string;
+    tags: Tag[];
+    type: "static_config";
+    upload_time: string | Date;
+};
+
+export type BlobListItem = {
+    blob_name: string;
+    blob_size: number;
+    blob_type: string;
+    id: string;
+    last_seen: string | Date;
+    type: "text_blob";
+    tags: Tag[];
+};
+
+export type Comment = {
+    author: string;
+    comment: string;
+    id: number;
+    timestamp: string | Date;
+};
+
+export type Share = {
+    access_reason: string;
+    access_time: string | Date;
+    group_name: string;
+    reason_type: string;
+    related_object_dhash: string;
+    related_object_type: ObjectType;
+    related_user_login: string;
+};
+
+export type Query = {
+    id: number;
+    name: string;
+    query: string;
+    type: ObjectType;
+};
+
+export type ApiKey = {
+    id: string;
+    issued_on: string | Date;
+    issuer_login: string;
+    name: string;
+};
+
+export type Family = {
+    count: number;
+    family: string;
+    last_upload: string;
+};
+
+export type KartonAnalyse = {
```

Comment:
I think this form would be better:
```suggestion
export type KartonAnalysis = {
```